### PR TITLE
Create README for Deno setup

### PR DESCRIPTION
**Pull request by bryophyta**
## What does this change?

Adds some documentation for setting up and using Deno, particularly in the context of managing github actions.

## Why?

Recent PRs have introduced Deno to the codebase for the first time.

## Screenshots

n/a

### CODE REVIEW

**Comment by mxdvl** [comment]
Agree with @jamesgorrie that this should be linked from the main `README.md`

**Comment by bryophyta** [comment]
> Agree with @jamesgorrie that this should be linked from the main `README.md`

Is this the sort of thing you had in mind @mxdvl @jamesgorrie? Or were you thinking about including all of the Deno content in the main README?
image

File: dotcom-rendering/README.md [hunk 1]
```diff
@@ -6,21 +6,21 @@ Frontend rendering framework for theguardian.com. It uses [React](https://reactj
...
+    -   [Architecture Diagram](#architecture-diagram)
+    -   [Concepts](#concepts)
+    -   [Feedback](#feedback)
+-   [Where can I see Dotcom Rendering in Production?](#where-can-i-see-dotcom-rendering-in-production)
+-   [Code Quality](#code-quality)
+    -   [Snyk Code Scanning](#snyk-code-scanning)
+-   [IDE setup](#ide-setup)
+    -   [Extensions](#extensions)
+    -   [Auto fix on save](#auto-fix-on-save)
+-   [Thanks](#thanks)
```
Comment (mxdvl) [inline dotcom-rendering/README.md:23]: Annoying that the formatting needs to change here.

File: dotcom-rendering/README.md [hunk 2]
```diff
@@ -75,17 +75,18 @@ If you're new to JavaScript projects, if you're trying to integrate with other a
...
+| <img alt="AB Testing" src="./docs/images/logo-ab-testing.png" width="350" />                                                                | The [A/B Testing library](https://github.com/guardian/ab-testing) is an internal NPM Module. There are a [some docs here](./docs/development/ab-testing-in-dcr.md).                                                                                                                                                                     |
+| <img alt="Deno" src="./docs/images/logo-deno.svg" width="350" /><p style="text-align: center;">[MIT License](https://deno.land/artwork)</p> | [Deno](https://deno.land/) is a JavaScript runtime that we've started incorporating into some of our Github Actions workflows. You will only need to install it if you are planning to run the workflow scripts locally. Some installation instructions can be found in the [Deno scripts folder](../scripts/deno/README.md).           |
```
Comment (mxdvl) [inline dotcom-rendering/README.md]: If we list Deno as a technology, we should also list Node for fairness.

**Review by mxdvl — APPROVED** [review]

File: dotcom-rendering/README.md (hunk 1 above)
Comment (bryophyta) [reply dotcom-rendering/README.md:23]: oh this was just a result of running Prettier, but if we don't want to use that for markdown files I'm happy to revert it -- I don't have strong opinions about formatting here

**Comment by mxdvl** [comment]
> Is this the sort of thing you had in mind @mxdvl @jamesgorrie? Or were you thinking about including all of the Deno content in the main README?

I think much like we have an [Install NodeJS](https://github.com/guardian/dotcom-rendering#install-nodejs) section, we should have a Github Actions / Install Deno section

**Comment by bryophyta** [comment]
> I think much like we have an [Install NodeJS](https://github.com/guardian/dotcom-rendering#install-nodejs) section, we should have a Github Actions / Install Deno section

Ah okay, my initial thinking was to keep the documentation fairly localised to avoid giving the impression that it's a core dependency which needs to be installed up-front, as I imagine that most contributors to the repo won't be touching the Actions scripts. Happy to add it elsewhere though, I'm still working on getting a feel for how the documentation is organised for this project.

If we're planning to add a CI/Github Actions section, maybe it makes sense to have a brief discussion of what we want to include in that?

File: dotcom-rendering/README.md (hunk 1 above)
Comment (mxdvl) [reply dotcom-rendering/README.md:23]: Fine!

File: dotcom-rendering/README.md [hunk 3]
```diff
@@ -75,17 +75,18 @@ If you're new to JavaScript projects, if you're trying to integrate with other a
...
+| <img alt="AB Testing" src="./docs/images/logo-ab-testing.png" width="350" />                                                                | The [A/B Testing library](https://github.com/guardian/ab-testing) is an internal NPM Module. There are a [some docs here](./docs/development/ab-testing-in-dcr.md).                                                                                                                                                                     |
+| <img alt="Deno" src="./docs/images/logo-deno.svg" width="350" />[MIT License](https://deno.land/artwork)| [Deno](https://deno.land/) is a JavaScript runtime that we've started incorporating into some of our Github Actions workflows. You will only need to install it if you are planning to run the workflow scripts locally. Some installation instructions can be found in the [Deno scripts folder](../scripts/deno/README.md).           |
```
Comment (mxdvl) [inline dotcom-rendering/README.md:89]: I think you can get rid of the `MIT License` comment.

**Review by mxdvl — APPROVED** [review]
Happy to get this merged in, and if we want to revisit in the future, we always can!

**Pull request merged**